modules/acquia_lift_publisher/src/EventSubscriber/Publish/PublishOnlyRendered.php

```diff
@@ -0,0 +1,165 @@
+
+  /**
+   * Acquia lift publisher configuration object.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   * @see \Drupal\acquia_lift_publisher\Form\AcquiaLiftPublisherSettingsForm
+   */
+  protected $publisherSettings;
+
+  /**
+   * PublishOnlyRendered constructor.
+   *
+   * @param \Drupal\acquia_contenthub\ContentHubCommonActions $common_actions
+   *   The Content Hub common actions service.
+   * @param \Drupal\Core\Config\ImmutableConfig $config
+   *   The publisher configuration object.
+   */
+  public function __construct(ContentHubCommonActions $common_actions, ImmutableConfig $config) {
+    $this->commonActions = $common_actions;
+    $this->publisherSettings = $config;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      ContentHubPublisherEvents::ENQUEUE_CANDIDATE_ENTITY => ['onEnqueueCandidateEntity', 100],
+      AcquiaContentHubEvents::PRUNE_PUBLISH_CDF_ENTITIES => ['onPrunePublishCdfEntities', 100],
+    ];
+  }
+
+  /**
+   * Checks if the entity in hand is a renderable entity.
+   *
+   * Only enqueue the entity if it is renderable.
+   *
+   * @param \Drupal\acquia_contenthub_publisher\Event\ContentHubEntityEligibilityEvent $event
+   *   The current enqueue eligibility event.
+   *
+   * @throws \Exception
+   */
+  public function onEnqueueCandidateEntity(ContentHubEntityEligibilityEvent $event): void {
+    if (!$this->personalizedContentPushIsActive()) {
+      return;
+    }
+
+    $entity = $event->getEntity();
+    $cdfs = $this->commonActions->getEntityCdf($entity);
+    foreach ($cdfs as $cdf) {
+      if ($cdf->getType() !== 'rendered_entity') {
+        continue;
+      }
+
+      $se_uuid = $this->getCdfEntityAttributeValue($cdf, 'source_entity');
+      if ($se_uuid === $entity->uuid()) {
+        $event->setEligibility(TRUE);
+        return;
+      }
+    }
+
+    $event->setEligibility(FALSE);
+  }
+
+  /**
+   * Removes cdfs that are not part of the personalized content scheme.
+   *
+   * @param \Drupal\acquia_contenthub\Event\PrunePublishCdfEntitiesEvent $event
+   *   The current publishing event.
+   */
+  public function onPrunePublishCdfEntities(PrunePublishCdfEntitiesEvent $event): void {
+    if (!$this->personalizedContentPushIsActive()) {
+      return;
+    }
+
+    $cdf_document = $event->getDocument();
+    $cdfs_to_remove = $cdf_document->getEntities();
+    $rendered_entity = NULL;
+    foreach ($cdfs_to_remove as $uuid => $cdf_entity) {
+      if ($cdf_entity->getType() === 'rendered_entity') {
+        $rendered_entity = $cdf_entity;
+        unset($cdfs_to_remove[$uuid]);
+        break;
+      }
+    }
+
+    $se_uuid = $this->getCdfEntityAttributeValue($rendered_entity, 'source_entity');
+    if (!$se_uuid) {
+      return;
+    }
+
+    $cdfs_to_keep = [$se_uuid];
+    $source_entity = $cdfs_to_remove[$se_uuid];
+    $tags = $this->getCdfEntityAttributeValue($source_entity, 'tags');
+    if (is_array($tags)) {
```

Comment:
I see, this is handled here. Perhaps you can ignore my previous comment.

Reply:
@kaynenh You can delete your comments, if you want.